akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -662,23 +661,6 @@ private[akka] class Shard(
       }
     case RestartRememberedEntities(entities) => restartEntities(entities)
     case l: LeaseLost                        => receiveLeaseLost(l)
-    case ShardRegion.StartEntityAck(entityId, _) =>
-      if (update.started.contains(entityId)) {
-        // currently in progress of starting, so we'll need to stop it when that is done
-        entities.rememberingStop(entityId, StartedElsewhere)
-      } else {
-        entities.entityState(entityId) match {
-          case _: RememberingStart =>
-            // queued up for batched start, let's just not start it
-            entities.removeEntity(entityId)
-          case _: Active =>
-            // add to stop batch
-            entities.rememberingStop(entityId, StartedElsewhere)
-          case _ =>
-            // not sure for other states, so deal with it later
-            stash()
```

Comment:
Same here

Reply:
I think this is where we'll end up. We're dong a update for another entity and an entity has moved. We need to stash it and then in idle we'll go through the remove logic so that it is removed from the remember store for this shard

Reply:
But as I said earlier, the other states should not be possible, because the scenario is:

1. remember entities was used
1. the cluster was stopped
1. he shard id extractor was changed 
1. when starting and remembering entities, they should run elsewhere, so the StartEntity goes there through the region

No message for those ids will be routed here because the extractor has changed, so the only possible state is `RememberingButNotStarted` here. Having an extra message per remembered entity also in the happy case just to cover that corner case seems wasteful, so either we keep it but let the RememberEntities actor filter them or introduce another more explicit message, I'm thinking maybe even collect those wrong ids into a separate message instead of one per entity.

WDYT @chbatey ?

Reply:
Implemented and pushed single message when RememberEntitiesStarter completes if any are relocated

Reply:
Yep, i like that. Way more explicit and more effecient

Reply:
But we also still need to remove them from our remember store to this doesn't happen over and over

Reply:
Ah, crap, I didn't think of that.

Reply:
At least we can batch that as one write now, better than before.

Reply:
Yep i think doing them all in one go is a good idea as if the user has changed their sharding functions there is likely a lot of moved entities